File: 42-trapping-rain-water/trapping-rain-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var trap = function (height) {
  if (height.length < 3) return 0;
  let left = 0;
  let right = height.length - 1;
  let maxLeft = height[left];
  let maxRight = height[right];
  let totalVol = 0;
  while (left < right) {
    if (maxLeft <= maxRight) {
      left++;
      maxLeft = Math.max(maxLeft, height[left]);
      totalVol += maxLeft - height[left];
    } else {
      right--;
      maxRight = Math.max(maxRight, height[right]);
      totalVol += maxRight - height[right];
    }
  }
  return totalVol;
};

File: 42-trapping-rain-water/trapping-rain-water.java
class Solution {
    public int trap(int[] height) {
            if (height.length < 3)
      return 0;
    int left = 0;
    int right = height.length - 1;
    int maxLeft = height[left];
    int maxRight = height[right];
    int totalVol = 0;
    while (left < right) {
      if (maxLeft <= maxRight) {
        left++;
        maxLeft = Math.max(maxLeft, height[left]);
        totalVol += maxLeft - height[left];
      } else {
        right--;
        maxRight = Math.max(maxRight, height[right]);
        totalVol += maxRight - height[right];
      }
    }
    return totalVol;
    }
}